File: baekjoon/정렬/11652.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;

    long long arr[n];

    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }

    sort(arr, arr+n);

    int cur_cnt = 1;
    int max_cnt = 1;
    long long max_num = arr[0];

    for (int i = 1; i < n; ++i) {
        if (arr[i] == arr[i-1]) cur_cnt++;
        else {
            if (cur_cnt > max_cnt) {
                max_cnt = cur_cnt;
                max_num = arr[i-1];
            }
            cur_cnt = 1;
        }
    }
    if (cur_cnt > max_cnt) max_num = arr[n-1];

    cout << max_num << '\n';

    return 0;
}

File: baekjoon/정렬/11652.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        List<Long> arr = new ArrayList<>();
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; ++i) {
            arr.add(Long.parseLong(br.readLine()));
        }

        arr.sort(Comparator.naturalOrder()); // 오름차순 정렬
        Long max_num = arr.get(0);
        int max_cnt = 1;
        int cur_cnt = 1;
        for (int i = 1; i < n; ++i) {
            if (arr.get(i).compareTo(arr.get(i-1)) == 0) {
                cur_cnt++;
            }
            else {
                if (cur_cnt > max_cnt) {
                    max_cnt = cur_cnt;
                    max_num = arr.get(i-1);
                }
                cur_cnt = 1;
            }
        }
        if (cur_cnt > max_cnt) {
            max_num = arr.get(n-1);
        }

        bw.write(max_num.toString());
        bw.write('\n');
        bw.flush();
        bw.close();
    }
}